sql/src/influxql/stmt_rewriter.rs

```diff
@@ -0,0 +1,460 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{collections::BTreeSet, ops::ControlFlow};
+
+use common_util::error::BoxError;
+use influxdb_influxql_parser::{
+    common::{MeasurementName, QualifiedMeasurementName, ZeroOrMore},
+    expression::{walk, Expr, WildcardType},
+    identifier::Identifier,
+    literal::Literal,
+    select::{
+        Dimension, Field, FieldList, FromMeasurementClause, MeasurementSelection, SelectStatement,
+    },
+};
+use itertools::{Either, Itertools};
+use snafu::{ensure, OptionExt, ResultExt};
+
+use super::{planner::MeasurementProvider, util};
+use crate::influxql::error::*;
+
+/// Rewriter for the influxql statement
```

Comment:
Those comment may should be placed at the beginning of file, and use `//!` syntax